contribs/gmf/src/services/authenticationservice.js

```diff
@@ -0,0 +1,263 @@
+goog.provide('gmf.Authentication');
+goog.provide('gmf.AuthenticationEventType');
+
+goog.require('gmf');
+goog.require('goog.events.EventTarget');
+goog.require('goog.uri.utils');
+
+
+/**
+ * @typedef {{
+ *     functionalities: (gmf.AuthenticationFunctionalities|undefined),
+ *     is_password_changed: (boolean|undefined),
+ *     role_id: (number|undefined),
+ *     role_name: (string|undefined),
+ *     username: (string|undefined)
+ * }}
+ */
+gmf.AuthenticationLoginResponse;
+
+
+/**
+ * @typedef {{
+ *     default_basemap: Array.<string>,
+ *     location: Array
+ * }}
+ */
+gmf.AuthenticationFunctionalities;
+
+
+/**
+ * @typedef {{
+ *     success: boolean
+ * }}
+ */
+gmf.AuthenticationDefaultResponse;
+
+
+/**
+ * @typedef {{
+ *     functionalities: (gmf.AuthenticationFunctionalities|null),
+ *     is_password_changed: (boolean|null),
+ *     role_id: (number|null),
+ *     role_name: (string|null),
+ *     username: (string|null)
+ * }}
+ */
+gmf.User;
+
+
+gmfModule.value('gmfUser', {
+  'functionalities': null,
+  'is_password_changed': null,
+  'role_id': null,
+  'role_name': null,
+  'username': null
+});
+
+
+
+/**
+ * An "authentication" service for a GeoMapFish application. Upon loading, it
+ * launches a request to determine whether a user is currently logged in or
+ * not.
+ *
+ * The possible API requests it supports, which are all self-explanatory, are:
+ *
+ * - changePassword
+ * - login
+ * - logout
+ * - resetPassword
+ *
+ * @constructor
+ * @extends {goog.events.EventTarget}
+ * @param {angular.$http} $http Angular http service.
+ * @param {string} authenticationBaseUrl URL to "authentication" web service.
+ * @param {gmf.User} gmfUser
+ * @ngInject
+ */
+gmf.Authentication = function($http, authenticationBaseUrl, gmfUser) {
+
+  goog.base(this);
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.$http_ = $http;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.baseUrl_ = authenticationBaseUrl;
+
+  /**
+   * @type {gmf.User}
+   * @private
+   */
+  this.user_ = gmfUser;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.changePasswordRouteSuffix_ = 'loginchange';
```

Comment:
How about we move them in a

gmf.AuthenticationRouteSuffix = {
  CHANGE_PASSWORD: 'loginchange',
  // etc...
}